src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java

```diff
@@ -51,7 +57,7 @@ public HarvestAgentClient(String host, int port) {
     }
 
     public String baseUrl() {
-        return host + ":" + port;
+        return "http://" + host + ":" + port;
```

Comment:
I think there might be more hard-coded 'http' literals in the code-base, but we'll have to clean this up eventually if we want to be able to use https between components.

Reply:
Accept. I would create an abstract client class and define the protocol scheme, host and port of a client.